vulnerabilities/api.py

```diff
@@ -210,8 +210,17 @@ def filter_purl(self, queryset, name, value):
                 detail={"error": f'"{purl}" is not a valid Package URL: {ve}'},
             )
 
-        attrs = {k: v for k, v in purl.to_dict().items() if v}
-        return self.queryset.filter(**attrs)
+        lookups = get_purl_query_lookups(purl)
+        return self.queryset.filter(**lookups)
+
+
+def get_purl_query_lookups(purl):
```

Comment:
Not a great location for this function. The model or around the QuerySet would be better.